src/core_modules/capture-core/components/WidgetEnrollment/TransferModal/hooks/useTransferValidation.js

```diff
@@ -0,0 +1,104 @@
+    const captureScopeQuery = dataEngine.query({
+        orgUnits: {
+            resource: 'organisationUnits',
+            params: {
+                paging: false,
+                userOnly: true,
+                fields: 'id,displayName',
+            },
+        },
+    });
+
+    const ancestorsQuery = dataEngine.query({
+        ancestors: {
+            resource: 'organisationUnits',
+            id: orgUnitId,
+            params: {
+                fields: 'ancestors[id,displayName]',
+            },
+        },
+    });
+
+    const result = await Promise.all([
+        captureScopeQuery,
+        ancestorsQuery,
+    ]);
+
+    const [{ orgUnits: { organisationUnits } }, { ancestors: { ancestors } }] = result;
+    ancestors.push({ id: orgUnitId });
+    return ancestors.some(({ id: ancestorId }) => organisationUnits.some(({ id }) => ancestorId === id));
+};
+
+export const useTransferValidation = ({ selectedOrgUnit, ownerOrgUnitId, programId }: Props) => {
+    const [validOrgUnit, setValidOrgUnit] = useState();
+    const [orgUnitScopes, setOrgUnitScopes] = useState({
+        ORIGIN: null,
+        DESTINATION: null,
+    });
+    const dataEngine = useDataEngine();
+    const { accessLevel } = useProgramAccessLevel({ programId });
+
+    useEffect(() => {
+        const updateOriginScope = (newScope: $Values<typeof OrgUnitScopes>) => setOrgUnitScopes(
+            prevOrgUnitScopes => ({
+                ...prevOrgUnitScopes,
+                ORIGIN: newScope,
+            }),
+        );
+
+        orgUnitIsInCaptureScope(ownerOrgUnitId, dataEngine).then((isInCaptureScope) => {
+            if (isInCaptureScope) {
+                updateOriginScope(OrgUnitScopes.CAPTURE);
+            } else {
+                updateOriginScope(OrgUnitScopes.SEARCH);
+            }
+        });
+    }, [dataEngine, ownerOrgUnitId]);
+
+    useEffect(() => {
+        if (!selectedOrgUnit) return;
+        const updateDestinationScope = (newScope: $Values<typeof OrgUnitScopes>) => setOrgUnitScopes(
+            prevOrgUnitScopes => ({
+                ...prevOrgUnitScopes,
+                DESTINATION: newScope,
+            }),
+        );
+
+        if (accessLevel === ProgramAccessLevels.OPEN || accessLevel === ProgramAccessLevels.AUDITED) {
+            updateDestinationScope(OrgUnitScopes.CAPTURE);
+        }
+        if (accessLevel === ProgramAccessLevels.PROTECTED || accessLevel === ProgramAccessLevels.CLOSED) {
+            orgUnitIsInCaptureScope(selectedOrgUnit?.id, dataEngine).then((isInCaptureScope) => {
+                if (isInCaptureScope) {
+                    updateDestinationScope(OrgUnitScopes.CAPTURE);
+                } else {
+                    updateDestinationScope(OrgUnitScopes.SEARCH);
+                }
+            });
+        }
+        setValidOrgUnit(selectedOrgUnit);
+    }, [accessLevel, dataEngine, selectedOrgUnit]);
```

Comment:
This hook is triggered when the `selectedOrgUnit` is the same (every time you click on the same org unit). Does it make sense to improve the dependency array checks to avoid unnecessary API calls made by `orgUnitIsInCaptureScope`? Thanks!

https://github.com/dhis2/capture-app/assets/6817073/4780d91d-a164-4766-9a2d-49edb3be6bed

Reply:
This makes sense! For some reason I was not able to reproduce on my own machine, but please let me know if the issue still persists. Rather than updating the dependancy-array, it made more sense to me to check this in the `handleSelectOrgUnit`-callback. I don't see any big reasons for actually updating the state if the orgUnit has not changed. Does that make sense?